[Bug]: patch file is not flushed correctly

### What happened?


Something is still wrong from  the patch Jesse provided for [aspect-cli#828](https://github.com/aspect-build/aspect-cli/issues/828) issue.

here is us doing `bazel lint //...`

```
(20:17:04) [10,377 / 10,458] Linting //ce-metrics:ce-metrics with Spotbugs; 8s disk-cache, darwin-sandbox ... (10 actions, 9 running)
(20:18:11) [10,442 / 10,458] Linting //core:core-neverlink with Spotbugs; 67s disk-cache, darwin-sandbox ... (2 actions, 1 running)
(20:18:18) INFO: Found 5164 targets...
(20:18:19) INFO: Elapsed time: 498.067s, Critical Path: 164.97s
(20:18:19) INFO: 6122 processes: 2384 disk cache hit, 175 internal, 3508 darwin-sandbox, 55 worker.
(20:18:19) INFO: Build completed successfully, 6122 total actions
(20:18:19) INFO:
Lint results for //clients:clients:

File fails SortImportOrder: clients/src/main/java/org/apache/kafka/common/requests/RequestContext.java

Lint results for //clients:clients-neverlink:

File fails SortImportOrder: clients/src/main/java/org/apache/kafka/common/requests/RequestContext.java

Linting failed due to errors.
make: *** [lint-check] Error 1
```

When I do` bazel lint //clients:clients` it passes :disappointed:

```
> bazel lint //clients:clients
(20:29:29) INFO: Current date is 2025-04-01
(20:29:29) Computing main repo mapping:
(20:29:29) Loading:
(20:29:29) Loading: 0 packages loaded
(20:29:29) Analyzing: target //clients:clients (0 packages loaded, 0 targets configured)
(20:29:29) Analyzing: target //clients:clients (0 packages loaded, 0 targets configured)

(20:29:29) INFO: Analyzed target //clients:clients (0 packages loaded, 0 targets configured).
(20:29:29) INFO: Found 1 target...
(20:29:29) INFO: Elapsed time: 0.267s, Critical Path: 0.00s
(20:29:29) INFO: 1 process: 1 internal.
(20:29:29) INFO: Build completed successfully, 1 total action
(20:29:29) INFO:
```

If I see the raw _exit_code_ it is **1** though

```
> bazel build //clients:clients  --aspects //tools/lint:linters.bzl%sort_imports --output_groups=rules_lint_machine --@aspect_rules_lint//lint:fix  --remote_download_all=true
(20:30:10) Loading: 0 packages loaded
(20:30:10) Analyzing: target //clients:clients (0 packages loaded, 0 targets configured)
(20:30:10) Analyzing: target //clients:clients (0 packages loaded, 0 targets configured)

(20:30:10) INFO: Analyzed target //clients:clients (0 packages loaded, 0 targets configured).
(20:30:10) INFO: Found 1 target...
Aspect //tools/lint:linters.bzl%sort_imports of //clients:clients up-to-date:
  bazel-bin/clients/clients.AspectRulesLintKeepJavaImportsSorted.report
  bazel-bin/clients/clients.AspectRulesLintKeepJavaImportsSorted.report.exit_code
(20:30:10) INFO: Elapsed time: 0.131s, Critical Path: 0.01s
(20:30:10) INFO: 2 processes: 1 disk cache hit, 1 internal.
(20:30:10) INFO: Build completed successfully, 2 total actions

> cat bazel-bin/clients/clients.AspectRulesLintKeepJavaImportsSorted.report.exit_code
1
```

I am using a new tag version I made: https://github.com/fzakaria/aspect-cli/commit/8c1bc1c9c634f7da26857c63f1446b6785f935ac


The **patch** file is **empty**

```
> bazel build //clients:clients  --aspects //tools/lint:linters.bzl%sort_imports --output_groups=rules_lint_patch --@aspect_rules_lint//lint:fi
x  --remote_download_all=true --@aspect_rules_lint//lint:debug=false --verbose_failures --sandbox_debug --output_groups=rules_lint_human
...
.
> cat bazel-out/darwin_arm64-fastbuild/bin/clients/clients.AspectRulesLintKeepJavaImportsSorted.patch
# nothing/empty
```

I did some more investigation and found the culprit.

This patch fixes it
```
--- a/lint/private/patcher.mjs
+++ b/lint/private/patcher.mjs
@@ -81,8 +81,9 @@ async function main(args, sandbox) {
     console.error(ret.error);
     process.exit(1);
   }
+

-  const diffOut = fs.createWriteStream(config.output);
+  const diffOut = fs.openSync(config.output, "w");
   const diffBin = process.env["DIFF_BIN"]
     ? path.join(process.env["JS_BINARY__RUNFILES"], process.env["DIFF_BIN"])
     : "diff";
@@ -101,7 +102,7 @@ async function main(args, sandbox) {
       }
     );
     debug(results.stdout);
-    diffOut.write(results.stdout);
+    fs.writeSync(diffOut, results.stdout);
     if (results.error) {
       console.error(results.error);
     }
```

I'm not very savvy with NodeJS but the async writing done earlier looks like it needs to handle buffering & sync which was not done.

This proved to be useful reference https://stackoverflow.com/q/43643467/143733

--- a/lint/private/patcher.mjs
+++ b/lint/private/patcher.mjs
@@ -81,8 +81,9 @@ async function main(args, sandbox) {
     console.error(ret.error);
     process.exit(1);
   }
+  
 
-  const diffOut = fs.createWriteStream(config.output);
+  const diffOut = fs.openSync(config.output, "w");
   const diffBin = process.env["DIFF_BIN"]
     ? path.join(process.env["JS_BINARY__RUNFILES"], process.env["DIFF_BIN"])
     : "diff";
@@ -101,7 +102,7 @@ async function main(args, sandbox) {
       }
     );
     debug(results.stdout);
-    diffOut.write(results.stdout);
+    fs.writeSync(diffOut, results.stdout);
     if (results.error) {
       console.error(results.error);
     }

